File: my-solutions/0217-contains-duplicate/solution.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        hashmap = {}
        hashmap[nums[0]] = 0
        for i in range(1, len(nums)):
            if nums[i] in hashmap.keys():
                return True
            else:
                hashmap[nums[i]] = i
        return False
        

File: my-solutions/0217-contains-duplicate/solution.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> num_set;
        for(int i = 0; i < size(nums); i++){
            if(num_set.find(nums[i]) != num_set.end()) {
                return true;
            } else {
                num_set.insert(nums[i]);
            }
        }
        return false;
    }
};
